### PR TITLE
Blocking cli command to run plan

### PR DESCRIPTION
**Pull request by RAYemelyanova**
The cli command to run a plan is now blocking.

### CODE REVIEW

**Comment by RAYemelyanova** [comment]
also Note: I've changed logging so that it is configured from the CLI not from the handler. Can take this out into its own PR if needed/desired

File: src/blueapi/cli/amq.py [hunk 1]
```diff
@@ -21,60 +13,49 @@ def __init__(self, message: str) -> None:
...
+    def wait_for_complete(self, timeout: Optional[float] = None) -> None:
+        begin_time = time.time()
+        timedout = False
+        while not self.complete.is_set():
+            current_time = time.time()
+            if (timeout is not None) and ((current_time - begin_time) > timeout):
+                timedout = True
+                break
+
+        if timedout:
```
Comment (callumforrester) [inline src/blueapi/cli/amq.py:56]: I believe the following will work, but test it:
```suggestion
        self.complete.wait(timeout=timeout)
        
        if not self.complete.is_set:
```

File: src/blueapi/cli/cli.py [hunk 2]
```diff
@@ -87,16 +94,47 @@ def get_devices(obj: dict) -> None:
 @controller.command(name="run")
 @click.argument("name", type=str)
 @click.argument("parameters", type=str, required=False)
+@click.option(
+    "-t",
+    "--timeout",
+    type=float,
+    help="Timeout for the plan in seconds. None hangs forever",
```
Comment (callumforrester) [inline src/blueapi/cli/cli.py:101]: I think we should word this in a more user friendly way given that hanging forever is a legitimate use case, we cannot predict how long some plans will take.
Maybe this...
```suggestion
    help="If supplied, terminates the client if the plan fails to finish within this time limit",
```

File: src/blueapi/messaging/context.py [hunk 3]
```diff
@@ -10,4 +10,4 @@ class MessageContext:
 
     destination: str
     reply_destination: Optional[str]
-    correlation_id: Optional[str]
+    task_id: Optional[str]
```
Comment (callumforrester) [inline src/blueapi/messaging/context.py:13]: This actually refers to header on the ActiveMQ message, which doesn't have anything to do with tasks etc. so it should be correlation_id and we shouldn't use it for now.

**Review by callumforrester — CHANGES_REQUESTED** [review]

**Pull request merged**